beesdoo_product/models/res_conig_settings.py

```diff
@@ -0,0 +1,41 @@
+# Copyright 2019-2020 Elouan Le Bars <elouan@coopiteasy.be>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class ResConfigSettings(models.TransientModel):
+    _inherit = "res.config.settings"
+
+    suggested_price_reference = fields.Selection(
+        selection=[
+            ("supplier_price", "Supplier Price"),
+            ("sale_price", "Sale Price"),
+        ],
+        string="Suggested price reference for margin",
+        help="""
+            Price on which the margin is applied when computing the suggested sale price. 
+            - Margin on Supplier Price : Suggested sale price = supplier price * (1 + margin / 100) (default)
+            - Margin on Sale Price: Suggested sale price = supplier price * (1 / (1 - margin / 100))
+        """,
+        default="supplier_price",
+    )
+
+    @api.model
+    def get_values(self):
+        res = super(ResConfigSettings, self).get_values()
+        select_type = self.env["ir.config_parameter"].sudo()
```

Comment:
having sudo activated at this level is something that I dislike... I rather prefer having it on function call for security reason.

Reply:
This is a standard block of code. It's like this, with the `sudo`, almost everywhere..

Reply:
It is not the main way of use. I know that in Odoo standard code we find it done that way sometimes mainly in res.config stuff but it's not recommended anywhere else.

I'm not against keeping it for res.config but a rather prefer standardization of use among our code. Maybe a point to discuss in coder council :)